File: phi_plugin/lib/GameUser.ts
import ByteReader from './ByteReader';
import Util from './Util';

export default interface GameUser {
    name: string;
    version: number;
    showPlayerId: boolean;
    selfIntro: string;
    avatar: string;
    background: string;
}

export default class GameUser {
    constructor(data: string) {
        this.name = "user";
        this.version = 1;
        let Reader = new ByteReader(data)
        this.showPlayerId = Util.getBit(Reader.getByte(), 0);
        this.selfIntro = Reader.getString();
        this.avatar = Reader.getString();
        this.background = Reader.getString();
    }
}

File: phi_plugin_js/lib/GameUser.py
from .ByteReader import ByteReader
from .Util import Util


class GameUser:
    """游戏用户"""

    name: str = "user"
    version: int = 1

    def __init__(self, data: bytes) -> None:
        """
        初始化

        :param data: 二进制数据
        """
        reader = ByteReader(data)
        self.showPlayerId = Util.getBit(reader.getByte(), 0)
        self.selfIntro = reader.getString()
        self.avatar = reader.getString()
        self.background = reader.getString()
